client/packages/common/src/ui/layout/tables/components/Header/ColumnPicker.tsx

```diff
@@ -55,30 +42,21 @@ export const ColumnPicker = <T extends RecordWithId>({
     setAnchorEl(null);
   };
 
-  const toggleColumn = (column: Column<T>) => {
-    const hidden = getHiddenColumns();
-    const updatedColumns = isVisible(column)
-      ? [...hidden, String(column.key)]
-      : hidden.filter(key => key !== column.key);
-
-    setHiddenColumnsConfig({
-      ...hiddenColumnsConfig,
-      [tableKey]: updatedColumns,
-    });
-  };
-
-  useEffect(() => {
-    onChange(columns.filter(isVisible));
-  }, [columns, onChange, isVisible]);
-
   return (
     <>
       <Tooltip title={t('table.show-columns')}>
         <IconButton onClick={handleClick} aria-describedby={id}>
           <ColumnsIcon
             sx={{
-              color:
-                getHiddenColumns().length > 0 ? 'secondary.main' : undefined,
+              color: !columns.every(
+                c =>
+                  columnDisplayState[String(c.key)] ||
+                  // If a column has no state (undefined), it must be a new
+                  // column, so we treat it as visible
+                  columnDisplayState[String(c.key)] === undefined
+              )
+                ? 'secondary.main'
+                : undefined,
```

Comment:
:nit this would be equivalent wouldn't it? To me it's easier to read, not requiring the negation on every. 

```suggestion
              // If a column has no state (undefined), it must be a new
              // column, so we treat it as visible
              color: columns.some(
                c => columnDisplayState[String(c.key)] === false
              )
                ? 'secondary.main'
                : undefined,
```